libiocage/lib/helpers.py

```diff
@@ -103,6 +103,82 @@ def validate_name(name):
     return bool(validate.fullmatch(name))
 
 
+# data -> bool | default
+def parse_bool(data, default=False):
+    """
+    try to parse booleans from strings
+
+    On success, it returns the parsed boolean
+    on failure it returns the `default`.
+    By default, `default` is `False`.
+
+    >>> parse_bool("YES")
+    True
+    >>> parse_bool("false")
+    False
+    >>> parse_bool("/opt/android-studio/build.txt")
+    False
+
+    Note that "-" gets a special treatment:
+
+    >>> parse_bool("-")
+    None
+
+    The behavior of the default parameter can be used to create a
+    pass-thru function:
+
+    >>> data = "/opt/android-studio/build.txt"
+    >>> parse_bool(data, data)
+    "/opt/android-studio/build.txt"
+    """
+
+    if isinstance(data, bool):
+        return data
+    if isinstance(data, str):
+        if data == "-":
+            return None
+        elif data.lower() in ["yes", "true", "on", "1"]:
+            return True
+        elif data.lower() in ["no", "false", "off", "0"]:
+            return False
+    return default
+
+
+def parse_user_input(data):
+    """
+    uses parse_bool() to partially return Boolean and NoneType values
+    All other types as returned as-is
+
+    >>> parse_bool("YES")
+    True
+    >>> parse_bool("false")
+    False
+    >>> parse_bool(8.4)
+    8.4
+    """
+    return parse_bool(data, data)
+
+
+def get_str_bool(data, style=None):
```

Comment:
Just an idea: We could also use two attributes `yes` and `no` containing the strings we expect for True or False values. like this:

```python
def get_str_bool(data, yes="yes", no="no"):
  ...

>>> get_str_bool(True, yes="yip", no="nope")
"yip"
>>> get_str_bool(False, yes="yip", no="nope")
"nope"
```

Reply:
As long as nobody invents a third boolean state for iocage... Personally I prefer the two parameters, because it seems more intuitive to me.